backend/icpy/api/rest_api.py

```
@@ -686,7 +808,31 @@ async def download_file(path: str):  # type: ignore
+                fs_service = self.filesystem_service
+                if self.context_router is not None:
+                    try:
+                        fs_service = await self.context_router.get_filesystem()
+                    except Exception:
+                        fs_service = self.filesystem_service
+                # Remote: stream bytes via adapter
+                if getattr(fs_service, 'is_remote', False):
+                    from fastapi.responses import StreamingResponse
+                    # For downloads, include a Content-Disposition filename
+                    filename = os.path.basename(path) or 'download'
+                    mime, _ = mimetypes.guess_type(filename)
```

Ruff (0.13.2)

Local variable `mimetypes` referenced before assignment

(F823)